src/wmtk/TriMesh.cpp

```diff
@@ -263,3 +317,47 @@ void TriMesh::swap_edge(const Tuple& t, int type)
 {
     throw "Not implemented";
 }
+
+std::vector<wmtk::TriMesh::Tuple> TriMesh::get_one_ring_tris_for_vertex(
+    const wmtk::TriMesh::Tuple& t) const
+{
+    std::vector<TriMesh::Tuple> one_ring;
+    size_t vid = t.get_vid();
+    auto conn_tri = m_vertex_connectivity[vid].m_conn_tris;
+    for (size_t tri : conn_tri) {
+        int j = m_tri_connectivity[tri].find(vid);
+        one_ring.emplace_back(vid, (j + 2) % 3, tri, *this);
+    }
+    return one_ring;
+};
+
+std::vector<wmtk::TriMesh::Tuple> TriMesh::get_one_ring_edges_for_vertex(
+    const wmtk::TriMesh::Tuple& t) const
+{
+    std::vector<Tuple> one_ring_edges;
+    wmtk::TriMesh::Tuple current_t = t;
+    int ring_size = m_vertex_connectivity[t.get_vid()].m_conn_tris.size();
+    for (int i = 0; i < ring_size; i++) {
+        wmtk::TriMesh::Tuple tmp_tuple = switch_face(t).value_or(current_t);
+        size_t fid = current_t.get_fid();
+        if (tmp_tuple.get_fid() < current_t.get_fid()) fid = tmp_tuple.get_fid();
+        int j = m_tri_connectivity[fid].find(current_t.get_vid());
+        one_ring_edges.emplace_back(current_t.get_vid(), (j + 2) % 3, fid, *this);
+        tmp_tuple = switch_edge(current_t);
+        current_t = tmp_tuple;
+    }
+    return one_ring_edges;
+};
+
+std::vector<wmtk::TriMesh::Tuple> TriMesh::get_incident_verts_for_tri(
```

Comment:
rename to `get_oriented_verts_for_tri`, see the corresponding function in `TetMesh.h`